### PR TITLE
[irods/irods#4740] Add iadmin modrepl and ls (master)

### PR DESCRIPTION
**Pull request by alanking**
iadmin modrepl can be used to modify a number of different columns in
R_DATA_MAIN. Some have been restricted until such a time as a valid use
case justifies its exposure in the client. Until then, only a subset of
the columns are modifiable via iadmin modrepl.

iadmin ls now lists the GenQuery attributes and their values which map to
the columns of R_DATA_MAIN. This is mainly for convenience.

---

CI tests running

### CODE REVIEW

**Comment by alanking** [comment]
CI tests passed.

File: src/iadmin.cpp [hunk 1]
```diff
@@ -1621,6 +1831,7 @@ void usageMain() {
         " rum (remove unused metadata (user-defined AVUs)",
         " asq 'SQL query' [Alias] (add specific query)",
         " rsq 'SQL query' or Alias (remove specific query)",
+        " modrepl logical_path [ replica_number <int> | resource <string> ] ATTR_NAME VALUE",
```
Comment (trel) [inline src/iadmin.cpp]: also modrepl data_id...

File: src/iadmin.cpp [hunk 2]
```diff
@@ -38,6 +39,269 @@ int lastCommandStatus = 0;
...
+    else if ("resource_hierarchy" == option) {
+        v = input.data();
+    }
+    return v;
+} // get_replica_value
+
+auto modify_replica(
+    char** tokens) -> int
+{
+    //const decltype(genquery_attrs)::key_type genquery_attrs_blacklist = {
```
Comment (trel) [inline src/iadmin.cpp]: remove comment

File: src/iadmin.cpp [hunk 3]
```diff
@@ -1582,11 +1793,10 @@ void usageMain() {
         " luan Name (list users associated with auth name (GSI/Kerberos)",
         " lt [name] [subname] (list token info)",
         " lr [name] (list resource info)",
-        " ls [name] (list directory: subdirs and files)",
+        " ls logical_path [ replica_number <int> | resource <string> ] (list replica info)",
```
Comment (trel) [inline src/iadmin.cpp]: also ls data_id

**Pull request merged**